python/ray/remote_function.py

```diff
@@ -116,14 +117,13 @@ def __init__(
             self._default_options["runtime_env"] = self._runtime_env
 
         self._language = language
-        self._function = _inject_tracing_into_function(function)
+        self._function = function
+        self._function_signature = None
+        self._inject_mut = Lock()
```

Comment:
nit: prefer `_inject_lock` to match type

also, add a small comment describing what this guards?